### ISSUE TITLE
Variable explorer don't show variables after connecting to remote kernel

### ISSUE THREAD

**Issue by 664787022** [spyder-ide/spyder#19548]
### Issue Report Checklist

- [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
- [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
- [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
- [x] Tried basic troubleshooting (if a bug/error)
    - [x] Restarted Spyder
    - [x] Reset preferences with ``spyder --reset``
    - [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    - [x] Tried the other applicable steps from the Troubleshooting Guide
- [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
Variable explorer don't show variables after connecting to remote kernel.

### What steps reproduce the problem?

1.  open a kernel with `python -m spyder_kernels.console` 
2. copy the json file to local dir
3. `connect to an exsiting kernel` in spyder

### What is the expected output? What do you see instead?
Variable explorer is expected to show variable `a` if i run 'a = 1'

### Paste Traceback/Error Below (if applicable)

```python-traceback

no traceback

```

## Versions

* Spyder version: 5.3.3  (conda)
* Python version: 3.9.13 64-bit
* Qt version: 5.15.4
* PyQt5 version: 5.15.7
* Operating System: Windows 10
* spyder-kernels version: 2.3.3

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                 :  1.4.1 (OK)
chardet >=2.0.0                      :  5.0.0 (OK)
cloudpickle >=0.5.0                  :  2.2.0 (OK)
cookiecutter >=1.6.0                 :  2.1.1 (OK)
diff_match_patch >=20181111          :  20200713 (OK)
intervaltree >=3.0.2                 :  3.0.2 (OK)
IPython >=7.31.1;<8.0.0              :  7.33.0 (OK)
jedi >=0.17.2;<0.19.0                :  0.18.1 (OK)
jellyfish >=0.7                      :  0.9.0 (OK)
jsonschema >=3.2.0                   :  4.16.0 (OK)
keyring >=17.0.0                     :  23.9.3 (OK)
nbconvert >=4.0                      :  7.0.0 (OK)
numpydoc >=0.6.0                     :  1.4.0 (OK)
paramiko >=2.4.0                     :  2.11.0 (OK)
parso >=0.7.0;<0.9.0                 :  0.8.3 (OK)
pexpect >=4.4.0                      :  4.8.0 (OK)
pickleshare >=0.4                    :  0.7.5 (OK)
psutil >=5.3                         :  5.9.2 (OK)
pygments >=2.0                       :  2.13.0 (OK)
pylint >=2.5.0;<3.0                  :  2.15.3 (OK)
pyls_spyder >=0.4.0                  :  0.4.0 (OK)
pylsp >=1.5.0;<1.6.0                 :  1.5.0 (OK)
pylsp_black >=1.2.0                  :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0            :  3.0.3 (OK)
qstylizer >=0.1.10                   :  0.2.2 (OK)
qtawesome >=1.0.2                    :  1.1.1 (OK)
qtconsole >=5.3.2;<5.4.0             :  5.3.2 (OK)
qtpy >=2.1.0                         :  2.2.0 (OK)
rtree >=0.9.7                        :  1.0.0 (OK)
setuptools >=49.6.0                  :  65.3.0 (OK)
sphinx >=0.6.6                       :  5.1.1 (OK)
spyder_kernels >=2.3.3;<2.4.0        :  2.3.3 (OK)
textdistance >=4.2.0                 :  4.5.0 (OK)
three_merge >=0.1.1                  :  0.1.1 (OK)
watchdog >=0.10.3                    :  2.1.9 (OK)
zmq >=22.1.0                         :  24.0.0 (OK)

# Optional:
cython >=0.21                        :  0.29.32 (OK)
matplotlib >=3.0.0                   :  3.6.0 (OK)
numpy >=1.7                          :  1.21.5 (OK)
pandas >=1.1.1                       :  1.5.0 (OK)
scipy >=0.17.0                       :  1.9.1 (OK)
sympy >=0.7.3                        :  1.10.1 (OK)

# Spyder plugins:
spyder_terminal.terminalplugin 1.2.2 :  1.2.2 (OK)

```

This issue seems to be resolved in #14894 , #281, #11538.  What happened to my spyder?

**Comment by mrclary** [comment]
@664787022, I cannot reproduce this issue with either our released 5.3.3 macOS standalone application or my experimental conda-based installer. But I'll try later with my Windows VM.

<img width="700" alt="Screen Shot 2022-09-30 at 2 53 45 PM" src="https://user-images.githubusercontent.com/9618975/193361783-755b474f-03cd-4c70-836f-25c2ed4a73ea.png">
<img width="700" alt="Screen Shot 2022-09-30 at 2 54 53 PM" src="https://user-images.githubusercontent.com/9618975/193361790-59b67c73-53ff-425a-a418-d4631d186c87.png">

**Comment by ccordoba12** [comment]
Thanks for checking @mrclary!

**Comment by mrclary** [comment]
I was unable to reproduce the OP in my Windows VM using the Windows standalone application, either. One thing that I did notice was that the variables didn't seem to show up in the Variable Explorer until I ran a command in the terminal. Although that did not seem to be consistent; other times the variables showed up immediately upon kernel connection (if the kernel already had variables). But it seemed that every time I actually created a new variable, it showed up immediately in Variable Explorer.
![issue-19548](https://user-images.githubusercontent.com/9618975/193377191-844d7b5c-af27-4234-9b44-fb60bde2bbea.gif)

**Comment by mrclary** [comment]
I also get the same behavior with our latest 5.x commit running from the command line conda environment. If the variables don't show up immediately upon kernel connection, clicking the refresh button in Variable Explorer immediately reveals them, also.

So, I don't know if there may be an issue with consistently refreshing the list at kernel connect, but otherwise I have not seen any issues displaying the workspace variables.

**Comment by 664787022** [comment]
> I also get the same behavior with our latest 5.x commit running from the command line conda environment. If the variables don't show up immediately upon kernel connection, clicking the refresh button in Variable Explorer immediately reveals them, also.
> 
> So, I don't know if there may be an issue with consistently refreshing the list at kernel connect, but otherwise I have not seen any issues displaying the workspace variables.

Thanks! I created a new environment and did not meet this problem again. I don't know what's going on with my last environment. But I can enjoy using spyder with remote server now. Thanks again!

**Comment by ccordoba12** [comment]
> So, I don't know if there may be an issue with consistently refreshing the list at kernel connect, but otherwise I have not seen any issues displaying the workspace variables.

Yeah, I think we should fix that to prevent confusions about the Variable Explorer not working, when it's really not displaying the current namespace state.

**Comment by driesdep** [comment]
![spyderVariableExplorereProbleem](https://github.com/spyder-ide/spyder/assets/81444996/246f07d6-e83a-4aff-822d-d344f0c540be)

I have the same problem; no variables ... Don't know how to fix this

**Comment by ccordoba12** [comment]
@driesdep, we're designing a new architecture to connect to remote machines and run code in them, which will be available in Spyder 6 (to be released in a couple of months).

So, please check after that version is out and let us know if the problem is still present then.
